crates/doc/src/shape/limits.rs

```diff
@@ -413,8 +430,28 @@ mod test {
             additionalProperties:
                 type: object
             "#,
-            &[doc],
-            Some(0),
+            docs,
+            Some((0, 99)),
+        );
+
+        widening_snapshot_helper(
+            None,
+            r#"
+            type: object
+            required: [bar9, foo9]
+            additionalProperties: false
+            properties:
+                bar9:
+                    type: object
+                    additionalProperties:
+                        type: object
+                foo9:
+                    type: object
+                    additionalProperties:
+                        type: object
+            "#,
+            docs,
+            Some((99, 1)),
```

Comment:
Perhaps add `Some((99, 2))` here as well, as a case?

I'm wondering if there are other ways we can bang on this. I do have high confidence in the change, but a mistake here would be rather painful to recover from.

Reply:
Yeah, good call. I added a couple more test cases here, including some that exercise arrays.